src/python/grpcio/grpc/experimental/aio/_call.py

```diff
@@ -88,8 +84,8 @@ def __init__(self,
         super().__init__(self)
         self._code = code
         self._details = details
-        self._initial_metadata = initial_metadata
-        self._trailing_metadata = trailing_metadata
+        self._initial_metadata = Metadata(*(initial_metadata or ()))
+        self._trailing_metadata = Metadata(*(trailing_metadata or ()))
```

Comment:
If Im not wrong most of the signatures that we have for returning metadata are `Optional[MetadataType]` [1] , meaning that if there is no metadata a `None` is returned. If this is the case every time that we build a metadata object we should do that if and only if there are values, having something like this:

```python
self._trailing_metadata = Metadata(*initial_metadata) if initial_metadata is not None else None
```

Also, note that maybe for avoiding the expansion - which might have a little impact - we could change the contractor of the `Metadata` for taking one parameter that would be the sequence.

[1] https://github.com/grpc/grpc/pull/23045/files#diff-802047f7d320089cb1d9c7de77c9cae5R107

Reply:
In the cases where it was returning ``None`` it's better to now return an empty metadata object now that we have the more specific type, as that will allow us to treat all calls to these methods the same (without exceptions of the kind ``if metadata is not None: ... ``), so I'd replace the ``Optional[MetadataType]`` by ``Metadata``. WDYT?

Reply:
For me I'm happy! WDYT @lidizheng | @gnossen ?

Reply:
Agree. Deterministic `Metadata` is better.